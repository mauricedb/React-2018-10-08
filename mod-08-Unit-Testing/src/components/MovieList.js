import React from 'react';
import propTypes from 'prop-types';

import MovieListItem from './MovieListItem';

const MovieList = ({ movies }) => (
  <div>
    <h1>Top  Movies</h1>
    {movies.map(movie => (
      <MovieListItem key={movie.id} movie={movie} />
    ))}
  </div>
);

MovieList.propTypes = {
  movies: propTypes.arrayOf(
    propTypes.shape({
      id: propTypes.number.isRequired
    })
  )
};

export default MovieList;
