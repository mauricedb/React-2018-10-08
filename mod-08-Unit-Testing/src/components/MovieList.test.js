import React from "react";
import reactTestRenderer from "react-test-renderer";
import MovieList from "./MovieList";

describe("The MovieList component", () => {
  it("renders the same tree with no movies", () => {
    const tree = reactTestRenderer.create(<MovieList movies={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders the same tree with a movie", () => {
    const tree = reactTestRenderer
      .create(
        <MovieList
          movies={[
            {
              id: 1,
              title: "A movie",
              backdrop_path: "the image url",
              overview: "The coool movie"
            }
          ]}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
