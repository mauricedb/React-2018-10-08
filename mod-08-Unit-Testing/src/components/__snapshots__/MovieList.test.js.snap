// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MovieList component renders the same tree with a movie 1`] = `
<div>
  <h1>
    Top  Movies
  </h1>
  <div>
    <h3
      onClick={[Function]}
    >
      A movie
    </h3>
    <img
      src="the image url"
    />
    <p>
      The coool movie
    </p>
  </div>
</div>
`;

exports[`The MovieList component renders the same tree with no movies 1`] = `
<div>
  <h1>
    Top  Movies
  </h1>
</div>
`;
